PyInstaller/hooks/hook-PyQt5.QtWebEngineWidgets.py

```diff
@@ -21,17 +22,16 @@
 if compat.is_darwin:
     # This is based on the layout of the Mac wheel from PyPi.
     data_path = pyqt5_library_info.location['DataPath']
-    resources = ['lib', 'QtWebEngineCore.framework', 'Resources']
-    web_engine_process = ['lib', 'QtWebEngineCore.framework', 'Helpers']
-    # When Python 3.4 goes EOL (see
-    # `PEP 448 <https://www.python.org/dev/peps/pep-0448/>`_, this is
-    # better written as ``os.path.join(*rel_data_path, *resources[:-1])``.
-    datas += collect_system_data_files(
-        os.path.join(data_path, *resources),
-        os.path.join(*(rel_data_path + resources[:-1])), True)
-    datas += collect_system_data_files(
-        os.path.join(data_path, *web_engine_process),
-        os.path.join(*(rel_data_path + web_engine_process[:-1])), True)
+    libraries = ['QtCore', 'QtWebEngineCore', 'QtQuick', 'QtQml', 'QtNetwork',
+                 'QtGui', 'QtWebChannel', 'QtPositioning']
+    for i in libraries:
+        datas += collect_system_data_files(
+            os.path.join(data_path, 'lib', i + '.framework'),
+            os.path.join(*(rel_data_path + ['lib'])), True)
+    datas += [(f, (os.path.basename(os.path.normpath(f))
+                   if os.path.isdir(f) else os.curdir)) for f
+                    in glob.glob(os.path.join(data_path, 'lib',
+                             'QtWebengineCore.framework', 'Resources', '*'))]
```

Comment:
This seems odd. Why the use of `os.curdir`? That seems likely to fail (eventually) when running tests in parallel.

Reply:
PyInstaller complains when a destination directory isn't passed. I used os.curdir for maximum compatibility instead of hard-coding a dot. However, I'm not sure what you mean about the tests: could you give me a link?

Reply:
I mean the Appveyor and travis tests that are run on every commit -- they run multiple tests in parallel. See `tests\functional\test_libraries.py`, for example.

Is the goal of this code to copy from `<path to Qt>/lib/QtWebEngineCore.framework/Resources` to `<built app>/lib/QtWebEngineCore.framework/Resources`? If so, I think that the existing code would work and is clearer (to me, at least):

``` Python
resources = ['lib', 'QtWebEngineCore.framework', 'Resources']
datas += collect_system_data_files(
    os.path.join(data_path, *resources),
    os.path.join(*(rel_data_path + resources[:-1])), True)
```

Reply:
Would this work, then?

``` Python
resources = ['lib', 'QtWebEngineCore.framework', 'Resources']
datas += collect_system_data_files(
    os.path.join(data_path, *resources),
    os.path.join('Resources'), True)
```

Mainly, I'm trying to avoid the use of `os.curdir` and provide a relative path as the destination.

Reply:
That actually results in the files getting put in `<built app>/Resources/Resources/Resources`. The main reason I didn't use `collect_system_data_files()` was because it actually put the files into `<built app>/Resources/Resources` (because the folder it was passed was called `Resources`).

Reply:
Ah, I see -- I'm not familiar with the Mac layout. Would this work?

``` Python
datas += collect_system_data_files(
    os.path.join(data_path, *resources),
    '..', True)
```

Reply:
Nope, already tried it. It causes a security alert because it contains `..`, even though it is still in a secure directory.

Reply:
Aargh. My goals for this code:

1. Find an alternative to `os.curdir`. @htgoebel, any idea on how to do this? `from PyInstaller.config import CONF; CONF['distpath']` still needs the name of the executable appended, but I'm not sure where to find that.

2. Replace the list comprehension with something more direct -- perhaps `datas += [(os.path.join(data_path, 'lib', 'QtWebengineCore.framework', 'Resources'), os.curdir)]`?

Reply:
Quick question -- why are we trying to avoid the use of `os.curdir`?

Reply:
To the best of my knowledge, it's value isn't guaranteed to be anything useful. I'm hoping to use a known-good reference to the filesystem such as `CONF['distpath']`, rather than assume future PyInstaller versions will set `os.curdir` to the `dist/` directory.

Reply:
The value of `os.curdir` seems to be hardcoded to `.`. PyInstaller "recommended" that I use a dot if I just wanted to assign these files to the root data directory (in this case `Bundle.app/Contents/MacOS/Resources`), and using `os.curdir` or a dot seems to works as intended (putting the files in the root data dir).